### PR TITLE
PYTHON-4045 Use PyPI Trusted Publishing

### PR DESCRIPTION
**Pull request by blink1073**
Here's a passing build on my fork: https://github.com/blink1073/mongo-python-driver/actions/runs/7023293310?pr=3

### CODE REVIEW

File: .github/workflows/release-python.yml [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+      - name: Build MacOS Py37 Wheel
+        if: ${{ matrix.python == 'cp37' && matrix.buildplat[0] == 'macos-11' }}
+        env:
+          CIBW_BUILD: cp37-macosx_x86_64
+          CIBW_ARCHS: x86_64
+          CIBW_TEST_COMMAND: "python {project}/tools/fail_if_no_c.py"
+        run: python -m cibuildwheel --output-dir wheelhouse
+
+      - name: Build wheels
+        if: ${{ matrix.python != 'cp37' || matrix.buildplat[0] != 'macos-11' }}
```
Comment (ShaneHarvey) [inline .github/workflows/release-python.yml:77]: Could you add a comment explaining why 3.7 and macos 11 skip this step?

File: .github/workflows/release-python.yml [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        - [ubuntu-20.04, manylinux_ppc64le]
+        - [ubuntu-20.04, manylinux_s390x]
+        - [ubuntu-20.04, manylinux_i686]
+        - [macos-11, macosx_*]
+        - [windows-2019, win_amd64]
+        - [windows-2019, win32]
+        python: ["cp37", "cp38", "cp39", "cp310", "cp311", "cp312"]
+
+    steps:
+      - name: Checkout pymongoarrow
```
Comment (ShaneHarvey) [inline .github/workflows/release-python.yml:42]: pymongoarrow -> pymongo

File: .github/workflows/release-python.yml [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+          # Build sdist on lowest supported Python
+          python-version: '3.7'
+
+      - name: Build SDist
+        run: |
+          set -ex
+          python -m pip install -U pip build
+          python -m build --sdist .
+
+      - name: Test Sdist
```
Comment (ShaneHarvey) [inline .github/workflows/release-python.yml:107]: Sdist -> SDist to match the naming above.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
Should we remove the Evergreen release tasks now? Or do we want to keep those as a backup?

File: .github/workflows/release-python.yml (hunk 1 above)
Comment (blink1073) [reply .github/workflows/release-python.yml:77]: Done

File: .github/workflows/release-python.yml (hunk 2 above)
Comment (blink1073) [reply .github/workflows/release-python.yml:42]: Fixed

File: .github/workflows/release-python.yml (hunk 3 above)
Comment (blink1073) [reply .github/workflows/release-python.yml:107]: Done

**Comment by blink1073** [comment]
> Should we remove the Evergreen release tasks now? Or do we want to keep those as a backup?

Yeah I intentionally left them as backup.

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**